app/src/main/java/org/p2p/wallet/history/ui/history/adapter/HistoryAdapterPagingController.kt

```diff
@@ -0,0 +1,35 @@
+package org.p2p.wallet.history.ui.history.adapter
+
+import org.p2p.wallet.common.ui.recycler.PagingState
+
+class HistoryAdapterPagingController(
+    private val adapter: HistoryAdapter
+) {
+    var currentPagingState: PagingState = PagingState.Idle
+
+    private companion object {
+        private val noAdditionalItemRequiredState = listOf(PagingState.Idle)
+    }
+
+    fun setPagingState(newState: PagingState) {
+        if (currentPagingState == newState) return
+
+        val shouldHasExtraItem = stateRequiresLoadingItem(newState)
+        val hasExtraItem = stateRequiresLoadingItem(currentPagingState)
+
+        currentPagingState = newState
+        // since item count is a function - cache its value.
+        val count = adapter.itemCount
+        when {
+            hasExtraItem && shouldHasExtraItem -> adapter.notifyItemChanged(count - 1)
+            hasExtraItem && !shouldHasExtraItem -> adapter.notifyItemRemoved(count - 1)
+            !hasExtraItem && shouldHasExtraItem -> adapter.notifyItemInserted(count)
+        }
+    }
+
+    fun stateRequiresLoadingItem(state: PagingState = currentPagingState): Boolean =
+        state !in noAdditionalItemRequiredState
+
+    fun isPagingInProgress(): Boolean =
+        currentPagingState is PagingState.Loading || currentPagingState is PagingState.InitialLoading
```

Comment:
We don't need `ProgressViewHolder` if state is `InitialLoading` 
We need progress element only if user loads second page or more

Reply:
so what should I do? it's not my check btw

<img width="699" alt="Screen Shot 2022-03-25 at 9 28 42 pm" src="https://user-images.githubusercontent.com/100791035/160180270-6685fcde-b5f8-4a55-8947-37b884784719.png">

Reply:
Yes, could you remove InitialLoading state here and add this element to `noAdditionalItemRequiredState` please